# Data Processing

In [5]:
from Agents import data_agent, badge_agent, report_agent, pre_game_report_agent
from openai import OpenAI
client = OpenAI(api_key= "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")

profile_pth = "Data/Lakers/Anthony_Davis/Profile.json"
asb_pth = "Data/Lakers/Anthony_Davis/Advanced_Score_Boxes.csv"
os_pth = "Data/Lakers/Anthony_Davis/Overall_Shooting.csv"
grs_pth = "Data/Lakers/Anthony_Davis/General_Range_Shooting.csv"
ds_pth = "Data/Lakers/Anthony_Davis/Dribbles_Shooting.csv"
pt_pth = "Data/Lakers/Anthony_Davis/Pass_To.csv"
pf_pth = "Data/Lakers/Anthony_Davis/Pass_From.csv"
d_pth =  "Data/Lakers/Anthony_Davis/Tracking_Defense.csv"

# Data Processing
player_profile, stats_1, stats_2, stats_3, stats_4, stats_5, stats_6, stats_7, stats_8 = data_agent.data_processing(profile_pth, asb_pth, os_pth, grs_pth, ds_pth, pt_pth, pf_pth, d_pth)

# Generate Player Badges

In [8]:
# Badge Generation
badge_user_input = badge_agent.create_badge_input(player_profile, stats_1, stats_2, stats_3, stats_4, stats_5, stats_6, stats_7, stats_8)
badge_result = badge_agent.generate_badge(client, badge_user_input)
print(badge_result)

Defense Badge  
Inside Scorer Badge  
Rebounding Badge


# Generate Player Report

In [9]:
# Report Generation
report_user_input = report_agent.create_report_input(player_profile, stats_1, stats_2, stats_3, stats_4, stats_5, stats_6, stats_7, stats_8)
report_result = report_agent.generate_report(client, report_user_input)
print(report_result)

**Pre-game Scouting Report: Anthony Davis**

**Overview:**
- **Name:** Anthony Davis
- **Number:** 3
- **Team:** Los Angeles Lakers
- **Position:** Power Forward/Center
- **Height:** 6'10" (2.08 meters)
- **Weight:** 253 lbs (115 kilograms)

**Player Average Stats: (Last Few Games)**
- **Points Per Game (PPG):** 31.08
- **Rebounds Per Game (RPG):** 11.17
- **Assists Per Game (APG):** 2.58
- **Steals Per Game (SPG):** 1.25
- **Blocks Per Game (BPG):** 2.0
- **Field Goal Percentage (FG%):** 57.19%
- **Three-Point Percentage (3P%):** 34.72%
- **Free Throw Percentage (FT%):** 79.38%

**Recent Performance Trends:**
Anthony Davis has been demonstrating strong performance on both ends of the court, contributing significantly to the Los Angeles Lakers' success in recent games, with a perfect winning streak except for a few close losses. He is currently on a hot streak with consistent high scoring and efficient shooting percentages.

**Role in the Team:**
Davis is a pivotal player for the Laker

# Generate Player Pre-Game Report

In [10]:
pre_game_report_user_input = pre_game_report_agent.create_pre_game_report_input(report_result,  badge_result)
pre_game_report_result = pre_game_report_agent.generate_pre_game_report(client, pre_game_report_user_input)

# Pre-Game Scouting Report

In [11]:
print(pre_game_report_result)

```markdown
<div style="text-align: center;">
  <span style="font-size: 32px; font-weight: bold;">Pre-Game Scouting Report</span>
</div>

<div style="margin: 20px 0;">
<img src="Data/Lakers/Anthony_Davis/Portrait.jpg" alt="Portrait" width="520" height="380" style="border: 1px solid #ddd; border-radius: 5px; padding: 5px;">
</div>

<div style="margin: 20px 0;">
  <img src="Badges/Defense.PNG" alt="Defense Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Finisher.PNG" alt="Inside Scorer Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Rebound.PNG" alt="Rebounding Badge" width="92" height="104" style="margin: 0 5px;">
</div>

<div>
  <span style="font-size: 32px; font-weight: bold;">Anthony Davis</span><br>
  <span style="font-size: 24px;">Los Angeles Lakers | 3 | Power Forward/Center</span>
</div>

<div style="margin: 30px 0;">
  <table style="width: 100%; border-collapse: collapse; text-align: center;">
    <thead style="background-color

<div style="text-align: center;">
  <span style="font-size: 32px; font-weight: bold;">Pre-Game Scouting Report</span>
</div>

<div style="margin: 20px 0;">
<img src="Data/Lakers/Anthony_Davis/Portrait.jpg" alt="Portrait" width="520" height="380" style="border: 1px solid #ddd; border-radius: 5px; padding: 5px;">
</div>

<div style="margin: 20px 0;">
  <img src="Badges/Defense.PNG" alt="Defense Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Finisher.PNG" alt="Inside Scorer Badge" width="92" height="104" style="margin: 0 5px;">
  <img src="Badges/Rebound.PNG" alt="Rebounding Badge" width="92" height="104" style="margin: 0 5px;">
</div>

<div>
  <span style="font-size: 32px; font-weight: bold;">Anthony Davis</span><br>
  <span style="font-size: 24px;">Los Angeles Lakers | 3 | Power Forward/Center</span>
</div>

<div style="margin: 30px 0;">
  <table style="width: 100%; border-collapse: collapse; text-align: center;">
    <thead style="background-color: #f2f2f2;">
      <tr>
        <th style="padding: 10px; border: 1px solid #ddd;">PPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">RPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">APG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">SPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">BPG</th>
        <th style="padding: 10px; border: 1px solid #ddd;">FG%</th>
        <th style="padding: 10px; border: 1px solid #ddd;">3P%</th>
        <th style="padding: 10px; border: 1px solid #ddd;">FT%</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td style="padding: 10px; border: 1px solid #ddd;">31.08</td>
        <td style="padding: 10px; border: 1px solid #ddd;">11.17</td>
        <td style="padding: 10px; border: 1px solid #ddd;">2.58</td>
        <td style="padding: 10px; border: 1px solid #ddd;">1.25</td>
        <td style="padding: 10px; border: 1px solid #ddd;">2.0</td>
        <td style="padding: 10px; border: 1px solid #ddd;">57.19%</td>
        <td style="padding: 10px; border: 1px solid #ddd;">34.72%</td>
        <td style="padding: 10px; border: 1px solid #ddd;">79.38%</td>
      </tr>
    </tbody>
  </table>
</div>

## Key Strengths:
- High shooting efficiency, particularly within 10 feet of the basket.
- Strong rebounder, utilizing size and athleticism with 11.17 RPG.
- Dominant interior defender with notable shot-blocking ability.

---

## Key Weaknesses:
- Inconsistent in three-point shooting, with limited attempts.
- Lower playmaking output at 2.58 APG.
- Prone to turnovers, averaging 2.5 per game.

---

## Offensive Strategy:
- Take advantage of perimeter gaps as Davis overcommits to interior defense.
- Utilize pick-and-roll plays to create mismatches and force defensive switches.

---

## Defensive Strategy:
- Double team in the paint to limit Davis' scoring opportunities.
- Encourage him to attempt long-range shots due to moderate 3P%.

Anthony Davis presents formidable challenges both offensively and defensively, requiring strategic planning to minimize his influence during the game.